developer-extension/src/panel/components/tabs/replayTab.tsx

```diff
@@ -43,30 +43,31 @@ function Player() {
   return (
     <TabBase>
       <iframe ref={frameRef} className={classes.iframe} data-status={playerStatus} />
-      {playerStatus === 'waiting-for-full-snapshot' && <WaitingForFullSnapshot />}
+      {playerStatus === 'waiting-for-full-snapshot' && <WaitingForFullSnapshot isReplayForced={isReplayForced} />}
     </TabBase>
   )
 }
 
-function WaitingForFullSnapshot() {
+function WaitingForFullSnapshot({ isReplayForced }: { isReplayForced: boolean }) {
   return (
     <Alert
       level="warning"
       message="Waiting for a full snapshot to be generated..."
       button={
-        <Button onClick={generateFullSnapshot} color="orange">
+        <Button onClick={() => generateFullSnapshot(isReplayForced)} color="orange">
           Generate Full Snapshot
         </Button>
       }
     />
   )
 }
 
-function generateFullSnapshot() {
+function generateFullSnapshot(isReplayForced: boolean) {
   // Restart to make sure we have a fresh Full Snapshot
+  const startParam = isReplayForced ? '{ force: true }' : ''
   evalInWindow(`
     DD_RUM.stopSessionReplayRecording()
-    DD_RUM.startSessionReplayRecording()
+    DD_RUM.startSessionReplayRecording(${startParam})
```

Comment:
**❓ question:** ‏Why is it needed? Couldn't we always use `force: true`?

Reply:
Yes we can, I just wanted to have the expected behaviour

Reply:
Done ✔️